libs/antsi/src/font.rs

```diff
@@ -0,0 +1,254 @@
+/// Font Weight
+///
+/// ## Support
+///
+/// Terminals that implement ANSI escape sequences, usually implement [`Bold`], but only some have
+/// support for [`Faint`]. This is often dependent on the available font-weights of the currently
+/// used terminal font.
+///
+/// ## Specification
+///
+/// The format for this escape sequence was standardized with the ANSI control sequences in
+/// [ISO 6429].
+///
+/// The escape sequences are:
+///
+/// | Font Weight           | Escape Code |
+/// |-----------------------|-------------|
+/// | [`FontWeight::Bold`]  | `ESC[1m`    |
+/// | [`FontWeight::Faint`] | `ESC[2m`    |
+///
+/// [ISO 6429]: https://www.iso.org/standard/12782.html
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum FontWeight {
+    /// Bold text
+    ///
+    /// Sometimes known as: increased intensity
+    // TODO: example
+    Bold,
+    /// Faint text
+    ///
+    /// Somtimes known as: decreased intensity, second color or light (CSS)
+    // TODO: example
+    Faint,
+}
+
+/// Alternative Font Family
+///
+/// [ISO 6429] only allows for up to 9 alternative fonts, therefore the value is clamped to `0..9`
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct AlternativeFontFamily(u8);
+
+impl AlternativeFontFamily {
+    pub fn new(variant: u8) -> Option<Self> {
+        (0..9).contains(&variant).then_some(Self(variant))
+    }
+}
+
+/// Font Family
+///
+/// ## Support
+///
+/// While part of the original specification [ISO 6429], a change of the font family is only
+/// supported on a select few of terminal emulators. Handle with care, as for most cases a change in
+/// font family won't change anything.
+///
+/// ## Specification
+///
+/// Support was part of the initial ANSI escape sequence specification [ISO 6429], the change of
+/// font families is supported by a few. Support for the [`Fraktur`] font is only available in a
+/// select few terminal emulators, like [Terminology](https://github.com/borisfaure/terminology).
+///
+/// The specification supports 9 alternative fonts.
+///
+/// [ISO 6429]: https://www.iso.org/standard/12782.html
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum FontFamily {
+    /// The primary (default) font family
+    Primary,
+    /// Fraktur
+    ///
+    /// What is Fraktur? [ISO 6429] was first introduced in 1976 as [ECMA-48], a time where in
```

Comment:
Needs reformatting to adjust line length:
```suggestion
    /// ## What is Fraktur?
    ///
    /// [ISO 6429] was first introduced in 1976 as [ECMA-48], a time where in
```